c/tests.c

```diff
@@ -730,6 +749,54 @@ test_take_ownership(void)
     }
 }
 
+static void
+test_borrow_array(void)
+{
+    int ret;
+    kastore_t store;
+    const uint32_t array[] = { 1, 2, 3, 4 };
+    uint32_t *a, *b, *c;
+    size_t array_len;
+    int type;
+
+    ret = kastore_open(&store, _tmp_file_name, "w", 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+
+    ret = kastore_puts(&store, "c", array, 4, KAS_UINT32, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = kastore_puts(&store, "b", array, 2, KAS_UINT32, KAS_BORROWS_ARRAY);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = kastore_puts(&store, "a", array, 1, KAS_UINT32, KAS_BORROWS_ARRAY);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
```

Comment:
We should add test for using one of the more specific forms too, to make sure the flags are passed through, e.g.kastore_puts_uint32